File: frontend/src/api/report.ts
import { apiRequest, type ApiFetch } from "./request";

export interface StandingRow {
  entry_id: string;
  points: number;
  rank: number;
}

export interface PodReport {
  is_complete: boolean;
  rounds_played: number;
  is_partial: boolean;
  standings: StandingRow[];
}

export function fetchPodReport(apiFetch: ApiFetch, podId: string): Promise<PodReport> {
  return apiRequest(apiFetch, `/pods/${podId}/report`);
}

File: backend/app/schemas/report.py
import uuid
from typing import Literal

from pydantic import BaseModel


class TiebreakValue(BaseModel):
    label: str
    value: float
    format: Literal["percent"]


class StandingRowRead(BaseModel):
    entry_id: uuid.UUID
    points: int
    rank: int
    tiebreakers: list[TiebreakValue]


class PodReport(BaseModel):
    is_complete: bool
    rounds_played: int
    is_partial: bool
    active_entry_count: int
    recommended_rounds: int
    standings: list[StandingRowRead]
